### ISSUE TITLE
Support for "remote" (EBS, NFS) file systems as data dir

### ISSUE THREAD

**Issue by guilload** [quickwit-oss/quickwit#3384]
Typically, users use `/qwdata` as a mount point for those file systems. Then, a Quickwit node believes the files stored in the data dir are local, and access is not shared. This creates some issues:
- multiple nodes are going to write at the same location, for instance in  `/qwdata/cache/splits`, `/qwdata/queues`, step on each other's toes, and eventually lose data and fail.
- caching files in the data dir (`SplitStore`, `MergeSplitDownloader`) no longer works
- actors time out because some operations take seconds instead of milliseconds.

**Comment by imotov** [comment]
I agree `/qwdata` feels a bit overcrowded at the moment with data that have very different and distinct purposes and lifecycles. We have:

1) **temp** - temporary node-specific data that are valid only for the duration of index or node lifetime what ever is shorter, that includes
    - `/qwdata/delete_task_service`
    - `/qwdata/indexing`
2) **cache** - temporary node-specific data that we would like to persist across node restarts but losing it will not lead to loss of data
    - `/qwdata/cache/splits`
3) **in-flight data** - temporary node-specific data that we have to persist across node restarts otherwise we will lose data in flight
    - `/qwdata/queues`
4) **persistent data** - persistent global data and meta-data we have to persist, users can already control its location.
    - `/qwdata/indexes` (default location for both splits and metastore)

Users can already control locations of 1-3) vs 4) and conveniently 1-3) are node-specific vs 4) that is global. So, I think nodes stepping on each other can be solved by adding a lock file or something similar. However, moving some of the directories to network storage just doesn't make sense. I think we should move 1) into TEMP directory, add ability for users to control location of 2) and 3) separately and add some documentation for the recommended locations for each of these groups in case of EBS/NFS deployments.

**Comment by ChangyouSiom** [comment]
> 
For example, I deploy a cluster with three Quickwit nodes on three machines, with one node on a single machine .    1-3）use local path and 4) use a shared NFS path.   Any risk for this deployment?

**Comment by guilload** [comment]
No, you're good. You've set up your nodes exactly as intended.

I don't think our documentation is very helpful today for EBS/NFS-based deployment, so how did you figure this out?

**Comment by ChangyouSiom** [comment]
Thanks a lot ~  I just knew about the uses for every paths, and carried out some tests. I'm worried about concurrent writing on index splits. Could you give me some details about this?

**Comment by fulmicoton** [comment]
Index splits are only written once and are then immutable. Their id itself is universally unique, so concurrent writes on a split cannot happen.

The metastore story however is a little bit more complicated. If you use postgresql everything should be alright.
If you are using the filebacked metastore, it is a specific JSON file also living in your indexes directory.
Concurrent writes would be a catastrophy. You will need to make sure that you have only one node running the metastore service.

Summing up:
- use postgresql as the metastore and you should be ok.
OR
- only start one node with the metastore service. It will handle all meta writes and read and you should be ok.
